tiled/adapters/jpeg.py

```diff
@@ -0,0 +1,279 @@
+                shape=arr.shape,
+                chunks=tuple((dim,) for dim in arr.shape),
+                data_type=BuiltinDtype.from_numpy_dtype(arr.dtype),
+            )
+        self._structure = structure
+
+    def metadata(self) -> JSON:
+        """
+
+        Returns
+        -------
+
+        """
+        return {}
+
+    def read(self, slice: Optional[NDSlice] = None) -> NDArray[Any]:
+        """
+
+        Parameters
+        ----------
+        slice :
+
+        Returns
+        -------
+
+        """
+        arr = np.asarray(self._file)
+        if slice is not None:
+            arr = arr[slice]
+        return arr
+
+    def read_block(
+        self, block: Tuple[int, ...], slice: Optional[slice] = None
+    ) -> NDArray[Any]:
+        """
+
+        Parameters
+        ----------
+        block :
+        slice :
+
+        Returns
+        -------
+
+        """
+        if sum(block) != 0:
+            raise IndexError(block)
+
+        arr = np.asarray(self._file)
+        if slice is not None:
+            arr = arr[slice]
+        return arr
+
+    def structure(self) -> ArrayStructure:
+        """
+
+        Returns
+        -------
+
+        """
+        return self._structure
+
+
+class JPEGSequenceAdapter:
+    """ """
+
+    structure_family = "array"
+
+    @classmethod
+    def from_uris(
+        cls,
+        data_uris: List[str],
+        structure: Optional[ArrayStructure] = None,
+        metadata: Optional[JSON] = None,
+        specs: Optional[List[Spec]] = None,
+        access_policy: Optional[AccessPolicy] = None,
+    ) -> "JPEGSequenceAdapter":
+        """
+
+        Parameters
+        ----------
+        data_uris :
+        structure :
+        metadata :
+        specs :
+        access_policy :
+
+        Returns
+        -------
+
+        """
+        seq = [path_from_uri(data_uri) for data_uri in data_uris]
+        return cls(
+            seq,
+            structure=structure,
+            specs=specs,
+            metadata=metadata,
+            access_policy=access_policy,
+        )
+
+    def __init__(
+        self,
+        seq: List[Path],
+        *,
+        structure: Optional[ArrayStructure] = None,
+        metadata: Optional[JSON] = None,
+        specs: Optional[List[Spec]] = None,
+        access_policy: Optional[AccessPolicy] = None,
+    ) -> None:
+        """
+
+        Parameters
+        ----------
+        seq :
+        structure :
+        metadata :
+        specs :
+        access_policy :
+        """
+        self._seq = seq
+        # TODO Check shape, chunks against reality.
+        self.specs = specs or []
+        self._provided_metadata = metadata or {}
+        self.access_policy = access_policy
+        if structure is None:
+            shape = (len(self._seq), *self.read(slice=0).shape)
+            structure = ArrayStructure(
+                shape=shape,
+                # one chunks per underlying TIFF file
```

Comment:
```suggestion
                # one chunks per underlying JPEG file
```
(although, unnecessary if we move this part to `FileSequenceAdapter`)